crates/time/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+            eras: vec![
+                Summary {
+                    start: Bound {
+                        time: 0,
+                        slot: 0,
+                        epoch: 0,
+                    },
+                    end: Bound {
+                        time: 86400000,
+                        slot: 86400,
+                        epoch: 1,
+                    },
+                    params: params.clone(),
+                },
+                Summary {
+                    start: Bound {
+                        time: 86400000,
+                        slot: 86400,
+                        epoch: 1,
+                    },
+                    end: Bound {
+                        time: 172800000,
+                        slot: 172800,
+                        epoch: 2,
+                    },
+                    params: params.clone(),
+                },
+            ],
+        };
+        let t0 = eras.slot_to_relative_time(172801);
+        assert_eq!(t0, Err(TimeHorizonError::PastTimeHorizon));
+        let t1 = eras.slot_to_relative_time(172800);
+        assert_eq!(t1, Ok(172800000));
+    }
+
+    #[test]
+    fn test_epoch_bounds() {
+        let params = EraParams::new(86400, 1000).unwrap();
+        let eras = EraHistory {
+            eras: vec![
+                Summary {
+                    start: Bound {
+                        time: 0,
+                        slot: 0,
+                        epoch: 0,
+                    },
+                    end: Bound {
+                        time: 864000000,
+                        slot: 864000,
+                        epoch: 10,
+                    },
+                    params: params,
+                },
+            ],
+        };
+        assert_eq!(eras.epoch_bounds(1).unwrap().start, 86400);
+        assert_eq!(eras.epoch_bounds(1).unwrap().end, 172800);
+        assert_eq!(eras.epoch_bounds(10), Err(TimeHorizonError::PastTimeHorizon));
+    }
+
+    #[test]
+    fn test_slot_to_epoch() {
+        let params = EraParams::new(86400, 1000).unwrap();
+        let eras = EraHistory {
+            eras: vec![
+                Summary {
+                    start: Bound {
+                        time: 0,
+                        slot: 0,
+                        epoch: 0,
+                    },
+                    end: Bound {
+                        time: 864000000,
+                        slot: 864000,
+                        epoch: 10,
+                    },
+                    params: params,
+                },
+            ],
+        };
+        let e0 = eras.slot_to_epoch(0);
+        assert_eq!(e0, Ok(0));
+        let e1 = eras.slot_to_epoch(86399);
+        assert_eq!(e1, Ok(0));
+        let e2 = eras.slot_to_epoch(864000);
+        assert_eq!(e2, Ok(10));
+        let e3 = eras.slot_to_epoch(864001);
+        assert_eq!(e3, Err(TimeHorizonError::PastTimeHorizon));
+    }
+
+    #[test]
+    fn test_encode_era_history() {
```

Comment:
One would typically want to have unit tests for encoding/decoding and a roundtrip property which is quite powerful to ensure whatever is encoded can be decoded and find edge cases.